File: Codes/Hash/Non-Cryptographic/RSHash/RSHash.pas

{
    Hash function by Robert Sedgwicks in C book.
    Archive of Reversing.ID
    Non-Cryptographic Hash
}

function RSHash (const key : String) : Cardinal;
const b = 378551;
var 
    a : Cardinal;
    i : Integer;
begin 
    a      := 63689;
    Result := 0;
    
    for i := 1 to Length(key) do
    begin
        Result := Result * a + Ord(key[i]);
        a      := a * b;
    end;
end;

File: Codes/Hash/Non-Cryptographic/RSHash/RSHash.java
/*
	Hash function by Robert Sedgwicks in C book.
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static long RSHash(String key)
{
    int b     = 378551;
    int a     = 63689;
    long state = 0;

    for(int i = 0; i < key.length(); i++)
    {
        state = state * a + key.charAt(i);
        a    = a * b;
    }

    return state;
}